javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -492,6 +492,19 @@ public boolean isEmpty() {
     }
 
     @Override
+    public <K2> TreeMap<K2, V> mapKeys(Function<? super K, ? extends K2> keyMapper) {
+        Objects.requireNonNull(keyMapper, "keyMapper is null");
+        return map((k, v) -> Tuple.of(keyMapper.apply(k), v));
+    }
+
+    @Override
+    public <K2> TreeMap<K2, V> mapKeys(Function<? super K, ? extends K2> keyMapper, Function2<V, V, V> valueMerge) {
+        //todo Hack
+        Comparator<K2> comparator = Comparators.naturalComparator();
```

Comment:
I think so.
For custom comparator we should create own methods with comparator as parameter